src/qtAliceVision/SequenceCache.cpp

```diff
@@ -260,55 +270,79 @@ void SequenceCache::onPrefetchingProgressed(int)
     Q_EMIT requestHandled();
 }
 
-void SequenceCache::onPrefetchingDone(int reqFrame)
+void SequenceCache::onPrefetchingDone(int sequenceId, int reqFrame)
 {
-    // Update internal state
-    _loading = false;
+    //Make sure the fetching concerns the actual sequence Id
+    bool exitOld = false;
+    _lockSequence.lock();
+    if (sequenceId != sequenceId)
+    {   
+        exitOld = true;
+    }
+    if (reqFrame >= _sequence.size())
```

Comment:
if the previous check passed then reqFrame should be in the correct range so this check is redundant (or maybe I missed something ?)

Reply:
ceinture et bretelle :-)